crates/astria-sequencer/src/accounts/state_ext.rs

```
@@ -0,0 +1,72 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use penumbra_storage::{
+    StateRead,
+    StateWrite,
+};
+
+pub type Address = String;
+pub type Balance = u64; // might need to be larger
+pub type Nonce = u32;
+
+const ACCOUNTS_PREFIX: &str = "accounts";
+
+fn storage_key(address: &str) -> String {
+    format!("{}/{}", ACCOUNTS_PREFIX, address)
+}
+
+pub(crate) fn balance_storage_key(address: &str) -> String {
+    format!("{}/balance", storage_key(address))
+}
+
+pub(crate) fn nonce_storage_key(address: &str) -> String {
+    format!("{}/nonce", storage_key(address))
+}
+
+#[async_trait]
+pub trait StateReadExt: StateRead {
+    async fn get_account_balance(&self, address: &str) -> Result<Balance> {
+        let bytes = self
+            .get_raw(&balance_storage_key(address))
+            .await
+            .expect("storage error");
```

Why panic instead of returning an error?